### ISSUE TITLE
🎨 Material 3️⃣ 

### ISSUE THREAD

**Issue by AndyScherzinger** [nextcloud/talk-android#2076]
EDIT: For any further details and discussions, please see PR  #2084

with the updates from #1982 and #1812 we are now able to implement Material 3
* https://material.io/blog/dynamic-color-harmony
* https://github.com/material-components/material-components-android/blob/master/docs/theming/Color.md#using-color-harmonization
---
<details>
<summary>Release notes 1.6.0</summary>
<p><em>Sourced from <a href="https://github.com/material-components/material-components-android/releases">material's releases</a>.</em></p>
<blockquote>
<h2>1.6.0</h2>
<h1>What's new since 1.5.0</h1>
<ul>
<li>Color harmonization support! Check out the following resources for more information:
<ul>
<li><a href="https://material.io/blog/dynamic-color-harmony">Designing Harmony into Dynamic Color</a> blog post</li>
<li><a href="https://github.com/material-components/material-components-android/blob/master/docs/theming/Color.md#using-color-harmonization">Using color harmonization</a> documentation</li>
</ul>
</li>
</ul>
<h2>Other highlights</h2>
<ul>
<li><code>Material 3</code>
<ul>
<li>Hide shadows on all M3 Toolbar styles, not just Surface style (b49284ec75ab16a9f649b97cb26aa4a17bc82512)</li>
<li>Migrate action bar popup themes to M3 (2ae3ca42985722900f53de9d9a1ef61c143767eb)</li>
<li>Update M3 card elevation according to the spec (bb23e2d6ad7888713d51e1dc53f36a69b0db62ff)</li>
<li>Add missing attributes to M3 button styles (3c9b45292d78a80f511d672bb5fe4ef1431dd8bd)</li>
<li>Update M3 navigation rail text appearances (6b57c52973c2ceb97c88da1df1b72859ed28775a)</li>
<li>Fix M3 date picker styles (fba1a341c31cbe2c5d71c239afc8bcc14af9d968)</li>
<li>Change checkedIconGravity values to enum (bc56d5b43b93ab6dbed4d561ad8b8b1351f9535c)</li>
</ul>
</li>
<li><code>Badging</code>
<ul>
<li>Prevent multiple badges being attached to a menu item at the same time (b3db1cc02855453f91353fca9e9cb11ccf0db9cc)</li>
<li>Refactored Badge state managing logic (bbbeacd64e3b1cc64e8edc6adf430aad45d03e32)</li>
<li>Added support for different locale on badges (b8f2dd53e0e19ea376639a610a0b51faabf0f9af)</li>
</ul>
</li>
<li><code>BottomNavigationView</code>
<ul>
<li>Added supported NavigationBarView#LabelVisibilityMode alternative to deprecated BottomNavigationView#LabelVisibilityMode. (12222684370dfb6c3400fd984578059d22741bb3)</li>
<li>Removed font scaling from NavigationBarView item labels. (d66676fd6bb93556c2cb3b0d9f7885821ac5bd8f)</li>
<li>Set backgroundTint explicitly to <a href="https://github.com/null"><code>@​null</code></a> in M3 bottom navigation styles (adbcf8cbbc6e95921d233c87eb87dc01b6cc8f0d)</li>
</ul>
</li>
<li><code>BottomSheet</code>
<ul>
<li>Fix bottom sheets in EXPANDED state when the expanded height is the same as the collapsed height (493243ea1e5a2a0eee675b652cae0eeaec73b6a3)</li>
<li>Simplify state transition logic (04c483cf3466cd0a7bf4835cadccc2372ba0da2b)</li>
<li>Fix activity leak when setting states on destroying activities (43114c4f7c9ca0c70c48c777dbd2957386d539e5)</li>
<li>Update edge-to-edge paddings when bottom sheet is laid out (19af0ac9d98cc3c504ce69ae38fbc00570cc85f5)</li>
</ul>
</li>
<li><code>Chip</code>
<ul>
<li>Refactored chip group selection logic to a standalone class (f3c6430f94bb7f82afc740e5545fd6302e9bf3aa)</li>
<li>Added missing attributes to M3 chip styles (cc18f3f7e8ff097b80d774911c2ede92d4b30698)</li>
<li>Update M3 chip styles with spec (8ed0ac9fd383bbb53fac100e74b2a0e9ee41bc9a)</li>
</ul>
</li>
<li><code>CollapsingToolbarLayout</code>
<ul>
<li>Fixed shadow issues when transition between expanded and collapsed states. (86e8b00ecdc9f9c5212e5a7e0700bd95be4da94b)</li>
<li>Constrain adjusted font weight to acceptable values (5dce006a5c1649df4eb6631c85619017728726af)</li>
</ul>
</li>
<li><code>Color</code>
<ul>
<li>Fix resources being re-harmonized when wrapContextIfAvailable() is used. (b402cb3c032ea10beb941284b49a9667ee7abe1e)</li>
<li>Added ColorGrid classes for color harmonization demo. (fcc10914f2b87c0e5f28761e8316bba2d9d1a318)</li>
<li>Add kyocera to dynamic colors allowlist and reorganized based on alphabetical order. (1dad5f65d4c54952a3a63efa4b9fd81b9c46d4a2)</li>
<li>Fix HarmonizedColorsOptions.Builder#setColorAttributes is marked as nullable (44054aaf3ab11d815e4086b37575e829e0b626db)</li>
<li>Added support for color resources harmonization in XML (869d9430677e66f9007bee65960dc0539627fd4a)</li>
<li>Refactor DynamicColors overloading methods and added DynamicColorsOptions class to support color harmonization. (bdbf052aaf1e824f221057ca3c3acae80d1a521d)</li>
<li>Update DynamicColors to use peekDecorView() instead of getDecorView() when applying dynamic color theme overlay (56ed07008ba0bff682e99c5d91126596fe9e6ddf)</li>
<li>Update DynamicColors util to applyStyle() on window decorView theme as well to fix potential ContextMenu issue (f6c8fa5aefc84a1543eca3056a538120de0b4297)</li>
<li>Update DynamicColors util to use activity.getTheme().applyStyle() instead of activity.setTheme() to workaround Force Dark issue (b84f35f1211028e7c456ce1ca91685a938f0ffcc)</li>
<li>Avoid Short#compare usage in ColorResourcesTableCreator.java since it requires API level 19 (48741b4e16236ee8ac9dc0052204e6a7be32c0f4)</li>
<li>Add Resource Table format for XML color Harmonization. (822a5b2cb2dedf133618d981140edf30db0058ef)</li>
</ul>
</li>
<li><code>Dialog</code>
<ul>
<li>Fixed crashes when colorControlHighlight is a CSL (1c5c054b70a55c2567cdfee460d3876158b2945a)</li>
<li>Added date and time picker themes to M3 dialog themes (8342f9025eefe64ca24312467d021a522e833769)</li>
</ul>
</li>
</ul>

</blockquote>
<p>... (truncated)</p>
</details>
---
<details>
<summary>Release notes 1.5.0</summary>
<p><em>Sourced from <a href="https://github.com/material-components/material-components-android/releases">material's releases</a>.</em></p>
<blockquote>
<h2>1.5.0</h2>
<h1>What's new since 1.4.0</h1>
<ul>
<li><code>Material3</code> themes, styles, and functionality! Check out the following resources for more information:
<ul>
<li><a href="https://m3.material.io">Material 3</a> site and guidelines</li>
<li><a href="https://material.io/blog/start-building-with-material-you">Start building with Material You</a> blog post</li>
<li><a href="https://material.io/blog/migrating-material-3">Migrating to Material Design 3</a> blog post</li>
<li><a href="https://github.com/material-components/material-components-android/blob/master/docs/getting-started.md">Getting started</a> &amp; <a href="https://github.com/material-components/material-components-android/blob/master/docs/theming/Color.md#using-dynamic-colors">dynamic color</a> developer documentation</li>
</ul>
</li>
<li>New <code>Divider</code> component (<a href="https://github.com/material-components/material-components-android/blob/master/docs/components/Divider.md">Docs</a>)</li>
</ul>
<h2>Other highlights</h2>
<ul>
<li><code>Badging</code>
<ul>
<li>Added support for differing offsets for badges with/without text, and support for badge width and padding in styles. (bd4914dd8bcdb64c4ec8734c95b90cec2a3d723d)</li>
</ul>
</li>
<li><code>BottomAppBar</code>
<ul>
<li>Added attr to set the color of navigation icon. (0ed7c7675e42cbe417fb947f150ed9faf6a5174b)</li>
<li>Added support for showing/hiding immediately (34740b6fdf164665c438658935653d354a64d72b)</li>
</ul>
</li>
<li><code>BottomNavigationView</code>
<ul>
<li>Updated to explicitly read and set minimum height. (943c4f0eedb35850f19167ac989387f5b640af2f)</li>
</ul>
</li>
<li><code>BottomSheet</code>
<ul>
<li>Added maxHeight to bottom sheet behavior (a38d2d89a8aa22a40dd200f2d1c9e837f87669c1)</li>
</ul>
</li>
<li><code>Chip</code>
<ul>
<li>Fixed icon tint states are not updated after setting new icon (c08a07d7e8247647ea939549c1a493f1d42ac326)</li>
</ul>
</li>
<li><code>CollapsingToolbarLayout</code>
<ul>
<li>Added option to add extra height when title text spans across multiple lines (655dde062aa4e7cd4cad0137f4e975c6ed1d2c27)</li>
<li>Added option to force always applying system window inset top regardless of layout_height (9ebf1a1387be4dfad00ad2af58ef9f02ec204d10)</li>
<li>Fixed multiline RTL collapsed title text position (af35dfb86bebdbd913e74df7cb828a0218b7889a)</li>
<li>Added support for expanded and collapsed title text color attributes (5c147722edcd5e6696248a1fdd372a1385d8a7df)</li>
<li>Fixed RTL text only laying out as RTL when actual text is RTL (841f2295921b4eed8bd16bb72b28d4ef7ca7cfa5)</li>
<li>Added title line spacing and hyphenation frequency setters (f0ce498108616ee26f84d8afd26c4627a4cdcf48)</li>
<li>Added support for multiline RTL when using fade title mode (f0ac3629682b003629858ab18bf74a878f2398ac)</li>
<li>Updated default multiline hyphenation frequency to StaticLayout.HYPHENATION_FREQUENCY_NORMAL (3ad232d829b8c1192b9c1af74225a69ac1632597)</li>
</ul>
</li>
<li><code>MaterialButton</code>
<ul>
<li>Updated Material Button style to set the preferred maximum width to 320dp. (c5c7a74cef46569ee74f92d0ef973cafe06c06e9)</li>
<li>Fixed MaterialButton lollipop shape issue (153fdee2f6f9f528856130be28c74f92bf739346)</li>
</ul>
</li>
<li><code>MaterialDatePicker</code>
<ul>
<li>Avoided NPE caused by getSelection() before created (cb5d622f131323bf1f27e9e6c27c16f4a918d3d1)</li>
<li>Fixed opening at the selected date in the end month (eb7b11478b110fc917958565876899856ba2411b)</li>
<li>Fixed DatePicker crashes and and potential issue of range selection (281688a2f21263817f037d187a98b4cbc8471985)</li>
<li>Fixed excessive whitespace (8fda897e1d2670b82752e8906a2a00700557408c)</li>
</ul>
</li>
<li><code>Snackbar</code>
<ul>
<li>Handled anchor view properly so no memory leak will happen (58ceeab63ca74d740fb96fbb5e999de325822330)</li>
<li>Fixed memory leak caused by not resetting anchor views (b0558dc006ac60f1c33cff4706bef109a6f77e07)</li>
<li>Fixed maxWidth and maxInlineActionWidth are not applied (8729d8ab571bc58582147cfb01c140d2c014617c, 15e2254397996e47128d73129b305abe5b17d071)</li>
</ul>
</li>
<li><code>TextInputLayout</code>
<ul>
<li>Applied tint when setting start icons (4044183f46385d09fe0059143bda77ce78f2dad9)</li>
<li>Made clear text icon focusable (8a4f42aca737f4b0c2b7e7851e557ef275df6ebd)</li>
<li>Added a fade transition to placeholder TextView's appear and disappear. (c92e6934d53032e9a3c0b4550787e665e6494d86)</li>
<li>Fixed cutout padding so text field outline doesn't overlap collapsed hint. (87b50c6aee240659841a7d5f2e52c1a6324d611f)</li>
</ul>
</li>
<li><code>TimePicker</code>
<ul>
<li>Hid cancel button when it's not cancelable (35c2af3c2dabdc470d55b8eddff56d174cbd9763)</li>
<li>Updated accessibility event sending when updating input mode (00c0df9e7d27c093b0b8ad82934a06637247bbae)</li>
</ul>
</li>
</ul>

</blockquote>
<p>... (truncated)</p>
</details>
